src/OnboardingSPA/components/Sidebar/components/Customize/DesignColorsPanel/index.js

```diff
@@ -81,6 +81,16 @@ const DesignColorsPanel = forwardRef(
 		}, [ currentData ] );
 
 		const colorPalettes = customizeSidebarData?.colorPalettes;
+
+		useEffect( () => {
+			if ( ! defaultGlobalData ) {
+				defaultGlobalData = cloneDeep(
+					currentData.sitegen.homepages.data
+				);
+			}
+			// eslint-disable-next-line react-hooks/exhaustive-deps
+		}, [ colorPalettes ] );
+
```

Comment:
We should avoid performing a `cloneDeep` every time `colorPalettes` are retrieved. This is a costly operation. Also, I think this will get duplicated with the `currentData` useEffect on lines 75-81. Let's remove these `useEffect`s along with `defaultGlobalData` and make `currentData.sitegen.homepages.active` the source of truth?